e2e/utils.go

```diff
@@ -310,3 +314,23 @@ func validateVariable(t *testing.T, contains bool, workingDir, name, value strin
 	}
 	assert.Fail(t, fmt.Sprintf("variable '%s' not found in terraform.tfvars", name))
 }
+
+func setParallelism(t *testing.T) {
```

Comment:
Nice! Appreciate the extra work to catch all of the parallel calls through out the test package.

You had mentioned the approach to use a test flag, and was wondering if you could share why it didn't work with the structure you have here?

```
var localFlag = flag.Bool("local", false, "")

func setParallelism(t *testing.T) {
    if !*local {
        t.Parallel()
    }
}
```

```
go test ./e2e -local
```

Reply:
I tried the `-short` flag but `testing.Short()` was always false and I didn't know why. I didn't dig too deep because I thought we might be overloading the usage of `-short` and adding some confusion anyway.
I wasn't aware that I can set custom flags, it is a cleaner approach for sure. I will test this and switch to it instead of using an env var if it works as expected.

Reply:
I see, it was around the `-short` option. I've typically paired `-short` with `t.Skip()`, but also haven't dug into the nuances with that case vs our parallel option. I like your callout that overloading that flag could add confusion, because that wouldn't make the testing run shorter, longer actually 😅 

Cool, interested to see how that plays out!

Reply:
Addressed in https://github.com/hashicorp/consul-terraform-sync/pull/636/commits/90228f263a6ea9ed79cf6ef524766175b82330c3